src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationRefresherProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    class AzureAppConfigurationRefresherProvider : IConfigurationRefresherProvider
+    {
+        public IEnumerable<IConfigurationRefresher> Refreshers { get; }
+
+        public AzureAppConfigurationRefresherProvider(IConfiguration configuration)
+        {
+            var configurationRoot = configuration as IConfigurationRoot;
+
+            if (configurationRoot == null)
+            {
+                throw new InvalidOperationException("Unable to access the Azure App Configuration provider. Please ensure that it has been configured correctly.");
+            }
+
+            var refreshers = new List<IConfigurationRefresher>();
+
+            foreach (IConfigurationProvider provider in configurationRoot.Providers)
+            {
+                if (provider is IConfigurationRefresher refresher)
```

Comment:
Did we want to throw if no refresher instance were found?

Reply:
Sure, we can throw `InvalidOperationException` in this case, similar to the case when we fail to downcast `IConfiguration` instance to `IConfigurationRoot`.

Reply:
Sounds good.